subcommand/health-sync/command.go

```diff
@@ -0,0 +1,181 @@
+package healthsync
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+)
+
+const (
+	flagContainerNames = "container-names"
+)
+
+type Command struct {
+	UI                 cli.Ui
+	flagContainerNames string
+	consulClient       *api.Client
+	log                hclog.Logger
+	flagSet            *flag.FlagSet
+	once               sync.Once
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagContainerNames, flagContainerNames, "", "Name of containers to sync to the service")
+	c.log = hclog.New(nil)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	// There is nothing to do, so successfully exit
+	if len(c.flagContainerNames) == 0 {
+		c.UI.Error(fmt.Sprintf("-%v doesn't have a value. exiting", flagContainerNames))
+		return 0
+	}
+
+	if err := c.realRun(); err != nil {
+		c.log.Error(err.Error())
+		return 1
+	}
+
+	return 0
+}
+
+func (c *Command) realRun() error {
+	c.ignoreSIGTERM()
+
+	consulClient, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return fmt.Errorf("constructing consul client: %s", err)
+	}
+	c.consulClient = consulClient
+
+	taskMeta, err := awsutil.ECSTaskMetadata()
+	if err != nil {
+		return err
+	}
+
+	// Duplicate code from mesh-init. Service ID must match here and in mesh-init.
+	serviceName := taskMeta.Family
+
+	currentStatuses := map[string]string{}
+	parsedContainerNames := strings.Split(c.flagContainerNames, ",")
+
+	for {
+		currentStatuses = c.syncChecks(currentStatuses, serviceName, parsedContainerNames)
+	}
+}
+
+func (c *Command) ignoreSIGTERM() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+	// Don't need to do anything for now. Just catch the SIGTERM so we don't exit.
+	// And, print when we receive the SIGTERM
+	go func() {
+		for sig := range sigs {
+			c.log.Info("signal received", "signal", sig)
+		}
+	}()
+}
+
+func (c *Command) syncChecks(currentStatuses map[string]string, serviceName string, parsedContainerNames []string) map[string]string {
```

Comment:
A comment on this function would be useful, to describe the health sync process and call out assumptions.